straticounter.m

```diff
@@ -1,123 +1,183 @@
+% parameter data sets (e.g. Vallelonga et al. (2014), Sigl et al. (in prep,
+% 2015)). For testing purposes, it can also be run on synthetic data.
+%
+% See Winstrup (2011) and Winstrup et al. (2012) for further documentation.
+%
+% When using this script, please provide release date of the algorithm,
+% and cite:
 % Winstrup et al., An automated approach for annual layer counting in
 % ice cores, Clim. Past. 8, 1881-1895, 2012.
-
-%% Copyright (C) 2015  Mai Winstrup
-% Files associated with the matchmaker software (matchmaker.m, 
-% matchmaker_evaluate.m) is authored and copyrighted by Sune Olander 
-% Rasmussen. 
-
-% This program is free software; you can redistribute it and/or modify it 
-% under the terms of the GNU General Public License as published by the 
-% Free Software Foundation; either version 2 of the License, or (at your 
+%
+%%%%%
+% Copyright (C) 2015  Mai Winstrup
+% Files associated with the matchmaker software (matchmaker.m,
+% matchmaker_evaluate.m) is authored and copyrighted by Sune Olander
+% Rasmussen.
+%
+% This program is free software; you can redistribute it and/or modify it
+% under the terms of the GNU General Public License as published by the
+% Free Software Foundation; either version 2 of the License, or (at your
 % option) any later version.
-
-% This program is distributed in the hope that it will be useful, but 
-% WITHOUT ANY WARRANTY; without even the implied warranty of 
-% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General 
+%
+% This program is distributed in the hope that it will be useful, but
+% WITHOUT ANY WARRANTY; without even the implied warranty of
+% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
 % Public License for more details.
-
+%
 % You should have received a copy of the GNU General Public License along
 % with this program; if not, write to the Free Software Foundation, Inc.,
 % 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-%% Check that settings file exist:
-if ~exist(['./Settings/' sett_icecore '.m'],'file')
-    disp('Settings file unknown, please correct')
-    return
+if ~isdeployed
+    % use the `isdeployed` variable to know whether this is a compiled instance
+    % of the code or not.
+
+    % If compiled we do not need to clc or close anything
+    clc; close all;
 end
 
+releasedate = '30-04-2015';
+
 %% Add paths to subroutines and settings folders:
-addpath(genpath('./Subroutines'))
-addpath(genpath('./Settings'))
+if ~isdeployed
+    addpath(genpath('./Subroutines'))
+    addpath(genpath('./Settings'))
+end
+
+%% Select model settings:
+% Import default settings:
+Model = defaultsettings();
+% Add release date:
+Model.releasedate = releasedate;
+
+% make generic message to be used when incorrect number of inputs are used
+vararg_err = 'This function accepts a maximum of two (2) input arguments';
+
+% use 'nargin' to see how many input arguments were used and select appropriate
+% behavior
+if nargin == 1
```

Comment:
```nargin``` checks to see how many input arguments there were. We do different things based on how many arguments were provided. You can see if there is only 1 input argument then we treat things the same as they used to be treated.